File: Interpreter/src/DataStructure/MyList.java
package DataStructure;


import ExceptionHandling.MyException;

import java.util.List;
import java.util.ArrayList;

public  class MyList<T> implements MyIntList<T> {

    List<T> list;

    public MyList(){
        list = new ArrayList<T>();
    }

    @Override
    public void add(T val) {
        list.add(val);
    }

    @Override
    public T get(int index) throws MyException {
        if(index < list.size() && index >= 0){
            return list.get(index);
        }
        else {
            throw new MyException("Index out of range!");
        }

    }

    @Override
    public void remove(int index) throws MyException {
        if(index < list.size() && index >= 0){
            list.remove(index);
        }
        else {
            throw new MyException("Index out of range!");
        }
    }

    @Override
    public int size() {
        return list.size();
    }

    @Override
    public String toString(){
        String printstr = new String();
        printstr += "Out";
        for(T e: list)
            printstr += "\n" + e.toString();
        return printstr;
    }
}

File: map/Interpreter2.0/DataStructure/MyList.cs
using System.Collections.Generic;
using MyException;
namespace DataStructure
{
    public class MyList<T> : MyIntList<T>
    {
    List<T> list;

    public MyList(){
        list = new List<T>();
    }

    public void add(T val) {
        list.Add(val);
    }

    public T get(int index) {
        if(index < list.Count && index >= 0){
            return list[index];
        }
        else {
            throw new MyException.MyException("Index out of range!");
        }

    }

    public void remove(int index){
        if(index < list.Count && index >= 0){
            list.RemoveAt(index);
        }
        else {
            throw new MyException.MyException("Index out of range!");
        }
    }

    public int size() {
        return list.Count;
    }


    override public string ToString(){
        string printstr = "";
        printstr += "Out";
        foreach(T e in list)
            printstr += "\n" + e.ToString();
        return printstr;
    }
        
    }
}